modules/apps/client-extension/client-extension-api/src/main/java/com/liferay/client/extension/util/CETUtil.java

```diff
@@ -5,14 +5,36 @@
 
 package com.liferay.client.extension.util;
 
+import com.liferay.client.extension.constants.ClientExtensionEntryConstants;
+import com.liferay.client.extension.model.ClientExtensionEntryRel;
+import com.liferay.client.extension.service.ClientExtensionEntryRelLocalServiceUtil;
 import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.model.LayoutSet;
+import com.liferay.portal.kernel.util.PortalUtil;
 import com.liferay.portal.kernel.util.Validator;
 
 /**
  * @author Binh Tran
  */
 public class CETUtil {
 
+	public static String getThemeCSSCETExternalReferenceCode(
```

Comment:
This made us think a lot. The other idea we had was to change the [getFrontendTokenDefinition from FrontendTokenDefinitionRegistry](https://github.com/liferay-platform-experience/liferay-portal/pull/460/commits/e65916b00fb8f4840b2d1827762e1147647ee507#diff-a59cab1fe870c51d4abbec837848440df17f9bb1fa723d7992d4dca9dd338634R13) to receive the `Layout` instead of the current three parameters `(long companyId, String externalReferenceCode, String themeId)`. That means `getThemeCSSCETExternalReferenceCode` would be a private method of `FrontendTokenDefinitionRegistryImpl`

Reply:
What do you think? @ethib137 Is the `FrontendTokenDefinitionRegistryImpl`- the right class to put it into? Idk, this sounds like something from `*CET*` class.

Reply:
We also have this method covered in a integration test. In case we need to change the parameter list to `Layout` as suggested, do you know any Util method which we can call to add and delete a `Layout`? so we can update our test as well

Reply:
> The other idea we had was to change the [getFrontendTokenDefinition from FrontendTokenDefinitionRegistry](https://github.com/liferay-platform-experience/liferay-portal/pull/460/commits/e65916b00fb8f4840b2d1827762e1147647ee507#diff-a59cab1fe870c51d4abbec837848440df17f9bb1fa723d7992d4dca9dd338634R13) to receive the Layout instead of the current three parameters (long companyId, String externalReferenceCode, String themeId). That means getThemeCSSCETExternalReferenceCode would be a private method of FrontendTokenDefinitionRegistryImpl

I would lean towards this... but I'm not a backend developer so tell me if this is a bad idea.

Reply:
@ethib137, since Frontend Infra is taking a look at this PR after us, let us them give a different perspective. What do you think?